CETCompression.md

```diff
@@ -368,6 +370,43 @@ def groupByIgnoringDigits(start: Long, end: Long, base: Int, numDigits: Int): Ve
 }
 ```
 
+## Adaptor Point Computation Optimizations
+
+Computing a large number of adaptor points for events with outcome composed of many digits is a computationally intensive process.
+To reduce the time and resources required, some optimizations can be applied.
+
+### Pre-computing
+
+The first optimization is to first compute the signature points for the possible value of each digit.
+For example, for an event with outcomes decomposed in base 2 with 10 digits, first compute:
+```
+S0_0 = R0 + 0 * P
+S0_1 = R0 + 1 * P
+S1_0 = R1 + 0 * P
+S1_1 = R1 + 1 * P
+...
+S9_0 = R9 + 0 * P
+S9_1 = R9 + 1 * P
+```
+
+Then compute the aggregate signature points by combining the Si_j.
+For example, to compute the signature point for the value `0 0 0 0 0 0 0 0 0 0`, compute:
+```
+S = S0_0 + S1_0 + ... + S9_0
+```
+
+### Memoization
+
+In addition to the above pre-computation, memoization can be used to further improve performance.
+Assuming an event with outcomes decomposed in base 2 with 10 digits, the algorithm works as follow:
+1. Compute `S0_0` for the value `0`
+2. Compute `S = S0_0 + S1_0` for the value `0 0`
+3. Compute `S' = S0_0 + S1_0 + S2_0` for the value `0 0 0`
+...
+10. Compute `S'' = S0_0 + S1_0 + ... + S8_0` for the value `0 0 0 0 0 0 0 0 0`
+11. Compute `S''' = S0_0 + S1_0 + ... + S9_0` for the value `0 0 0 0 0 0 0 0 0 0`
+12. Compute `S'''' = S'' + S9_1` for the value `0 0 0 0 0 0 0 0 0 1`
```

Comment:
This seems like more of an example than an algorithm imo

Reply:
Added TS code (thought rust was maybe not the most readable)

Reply:
honestly I think it might be nice to have the example first so that it is easier to make sense of the algorithm but otherwise looks fine to me (will of course look more closely later when I implement this)

Reply:
Good point updated to have the example first.  I don't think this is something urgent to merge so we can maybe wait until you have it implemented (and I cleaned up my implementation :) )